Engine/src/Scene/Component.h

```diff
@@ -74,7 +74,9 @@ class ComponentPool final : public IComponentPool {
     ComponentType& AddComponent(Entity entity, Args&&... args)
     {
         m_entityToComponentIndex.insert({ entity, m_components.size() });
-        m_components.emplace_back(std::forward<Args>(args)...);
+        ComponentType component { std::forward<Args>(args)... };
```

Comment:
This was very annoying - emplace_back here caused my compiler to spit out unreadable errors complaining about "construct_at" internal method, and according to some info online, this may be a bug in compiler itself or MacOS SDK. So had to use the old way here.